model/src/pyrenew/deterministic/deterministicpmf.py

```diff
@@ -46,7 +53,12 @@ def __init__(
             tol=tol,
         )
 
-        self.basevar = DeterministicVariable(vars, name)
+        self.basevar = DeterministicVariable(
+            vars=vars,
+            name=name,
+            t_start=t_start,
+            t_unit=t_unit,
+        )
```

Comment:
Beyond the scope of this PR but I think this is most efficiently handled by actual inheritance, e.g. something like:

```python
def __init__(
       self, 
       name: str
       pmf: ArrayLike,
       tol: float = 1e-5,
       **kwargs):
    normed_pmf = validate_discrete_dist_vector(pmf, tol)
    super().__init__(vars=normed_pmf, **kwargs)
```
This avoids having to override the `sample()` method, among other things.

Reply:
Smells like a wider discussion, if we consider the current preference of using composition instead of inheritance when building models.